app/scripts/components/exploration/timeline.tsx

```diff
@@ -0,0 +1,472 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
```

Comment:
This file is getting on the larger side. Maybe consider starting with extracting away `Controls` and `DatasetList`?